File: C-Programming/Competitive/Assignment_40_3.c
#include <stdio.h>
#include <stdlib.h>

struct node
{
   int data;
   struct node *next;
};

typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

int LastOccurance(PNODE first, int iNo)
{
   PNODE temp = NULL;
   int iCount = 0;
   int lastPos = 1;
   while (temp != NULL)
   {
      if(first->data == iNo)
      {
         lastPos = iCount++;
      }
      temp = temp->next;
   }
   iCount++;

   return iCount;
   
}

void Display(PNODE first)
{
   while (first != NULL)
   {
      printf("| %d | -> ", first->data);
      first = first->next;
   }
   printf("NULL\n");
}

void InsertFirst(PPNODE first, int iNo)
{
   PNODE newn = NULL;

   newn = (PNODE)malloc(sizeof(NODE));

   newn->data = iNo;
   newn->next = NULL;

   if(*first == NULL)
   {
      *first = newn;
   }
   else
   {
      newn->next = *first;
      *first = newn;
   }
}

int main()
{
   PNODE head = NULL;
   int iRet = 0, iValue = 0;

   InsertFirst(&head,21);
   InsertFirst(&head,11);
   InsertFirst(&head,54);
   InsertFirst(&head,21);
   InsertFirst(&head,16);

   Display(head);

   printf("Enter a number : ");
   scanf("%d", &iValue);

   iRet = LastOccurance(head, iValue);

   printf("Last occurrence of %d is at position %d\n", iValue, iRet);

   return 0;
}

File: Data-Structure/Assignment_40_3.java
import java.util.*;

class Node
{
   public int data;
   public Node next;
   
   public Node(int Value)
   {
      this.data = Value;
      this.next = null;
   }
}

class SinglyLinkedList
{
   private Node first;
   
   public SinglyLinkedList()
   {
      this.first = null;
   }
   
   public void InsertLast(int iNo)
   {
      Node newn = new Node(iNo);
      if(first == null)
      {
         first = newn;
      }
      else
      {
         Node temp = first;
         while(temp.next != null)
         {
            temp = temp.next;
         }
         temp.next = newn;
      }
   }

   public void Display()
   {
      Node temp = first;
      while(temp != null)
      {
         System.out.print("| " + temp.data + " | -> ");
         temp = temp.next;
      }
      System.out.println("NULL");
   }

   public int LastOccurance(int iNo)
   {
      Node temp = first;
      int iCount = 0, lastposition = 0;

      iCount = 1;
      while(temp != null)
      {
         if(temp.data == iNo)
         {
            lastposition = iCount;
         }
         temp = temp.next;
         iCount++;
      }
      return lastposition;
   }
}

class Assignment_40_3
{
   public static void main(String A[])
   {
      int iValue = 0, iRet = 0;

      SinglyLinkedList ssobj = new SinglyLinkedList();

      ssobj.InsertLast(21);
      ssobj.InsertLast(10);
      ssobj.InsertLast(51);
      ssobj.InsertLast(21);
      ssobj.InsertLast(111);

      ssobj.Display();

      Scanner sobj = new Scanner(System.in);

      System.out.println("Enter the number : ");
      iValue = sobj.nextInt();

      iRet = ssobj.LastOccurance(iValue);

      System.out.println("The Last occurance of the number is at position: "+iRet);

      sobj.close();
   }
}